packages/fuels-contract/src/script_calls.rs

```diff
@@ -122,14 +147,38 @@ where
     /// it will be a bool, works also for structs thanks to the `abigen!()`).
     /// The other field of [`FuelCallResponse`], `receipts`, contains the receipts of the transaction.
     async fn call_or_simulate(&self, simulate: bool) -> Result<FuelCallResponse<D>, Error> {
+        let contract_ids: HashSet<ContractId> = self
+            .script_call
+            .external_contracts
+            .iter()
+            .map(|bech32| bech32.into())
+            .collect();
+        let num_of_contracts = contract_ids.len();
+
+        let inputs = chain!(
+            generate_contract_inputs(contract_ids),
+            self.script_call.inputs.clone(), // TODO(iqdecay): allow user to set inputs field
```

Comment:
Can we track this with an issue instead?

Reply:
@iqdecay is this still relevant ?